src/handlers/wire.rs

```diff
@@ -70,46 +48,39 @@ pub mod aggregation {
 
     impl Write for Signature {
         fn write(&self, buf: &mut impl BufMut) {
-            (self.signature.len() as u32).write(buf);
-            buf.put_slice(&self.signature);
+            self.signature.write(buf);
         }
     }
 
     impl Read for Signature {
         type Cfg = ();
 
         fn read_cfg(buf: &mut impl Buf, _: &()) -> Result<Self, Error> {
-            let len = u32::read(buf)? as usize;
-            if buf.remaining() < len {
-                return Err(Error::EndOfBuffer);
-            }
-            let mut signature = vec![0u8; len];
-            buf.copy_to_slice(&mut signature);
+            let signature = Vec::<u8>::read_range(buf, ..)?;
             Ok(Self { signature })
         }
     }
 
     impl EncodeSize for Signature {
         fn encode_size(&self) -> usize {
-            4 + self.signature.len() // u32 for length + bytes
+            self.signature.encode_size()
         }
     }
 
     /// Defines the different types of messages exchanged during the aggregation protocol.
     #[derive(Clone, Debug, PartialEq)]
     pub enum Payload {
         /// Message sent by orchestrator to start aggregation
-        Start(Start),
+        Start,
         /// Sent by signer to all others
         Signature(Signature),
```

Comment:
You could also probably delete the `Signature` struct and use `Vec<u8>` directly :)